src/tools/fuzzing/fuzzing.cpp

```diff
@@ -4881,7 +4887,29 @@ Expression* TranslateToFuzzReader::makeRefCast(Type type) {
       // This unreachable avoids a warning on refType being possibly undefined.
       WASM_UNREACHABLE("bad case");
   }
-  return builder.makeRefCast(make(refType), type);
+  // Descriptor casts emit a type that depends on the descriptor, so we can only
+  // create one if that type would fit |type| which is what we must emit at the
+  // end.
+  Expression* descRef = nullptr;
+  if (auto desc = refType.getHeapType().getDescriptorType();
```

Comment:
The descriptor type is tied to the cast output type, so I believe this could be `auto desc = type.getHeapType().getDescriptorType()`. Then we wouldn't have to check that `refType <: type` below and could emit it no matter what the relationship between `type` and `refType`.